src/web/extractors.rs

```diff
@@ -2,19 +2,61 @@
 //!
 //! Handles ensuring the header's, body, and query parameters are correct, extraction to
 //! relevant types, and failing correctly with the appropriate errors if issues arise.
-use actix_web::{Error, FromRequest, HttpRequest, Path, State};
+use std::error;
+use std::fmt;
+use std::str::FromStr;
+
+use actix_web::http::StatusCode;
+use actix_web::{Error, FromRequest, HttpRequest, HttpResponse, Path, Query, ResponseError, State};
 use futures::{future, Future};
-use serde::de::{Deserialize, Deserializer};
+use num::Zero;
+use regex::Regex;
+use serde::de::{Deserialize, Deserializer, Error as SerdeError};
+use serde_json::from_value;
+use validator::{Validate, ValidationError, ValidationErrors};
 
-use db::{params, Db};
 use server::ServerState;
 use settings::Settings;
-use web::auth::{HawkIdentifier, HawkPayload};
+use web::auth::HawkIdentifier;
+
+const BATCH_MAX_IDS: usize = 100;
```

Comment:
Long term, I guess we want to read the value of this from `Settings`, in case it's been set to something different to the default value?

Reply:
Most likely, yea.